dashboard/cron.py

```diff
@@ -403,14 +418,66 @@ def update_term(self):
 
         return status
 
+
+    # update course records with data returned from verify_course_ids, only making changes when necessary
+    def update_course(self, warehouse_courses_data):
+        # cron status
+        status = ""
+        logger.debug("while using verify_course_ids data to update course table")
+
+        logger.info(warehouse_courses_data.to_json())
+        courses = Course.objects.all()
+        courses_string = ", ".join([str(x) for x in Course.objects.get_supported_courses()])
+        status += f"{str(len(courses))} course(s): {courses_string}\n"
+
+        for course in courses:
+            updated_fields = []
+            status += f"course {course.id}: updated "
+            warehouse_course_df = warehouse_courses_data.loc[warehouse_courses_data["id"] == course.id]
+
+            warehouse_course_name = warehouse_course_df["name"].iloc[0]
+            if course.name != warehouse_course_name:
+                course.name = warehouse_course_name
+                logger.info(f"Name for {course.id} has been updated.")
+                updated_fields.append("name")
+
+            # term is always updated because the term records are deleted by the update_term method
+            course.term = AcademicTerms.objects.get(id=warehouse_course_df["enrollment_term_id"])
+            logger.info(f"Term for {course.id} has been updated.")
+            updated_fields.append("term")
+
+            warehouse_course_start = warehouse_course_df["start_at"].iloc[0]
```

Comment:
The sections of warehouse_course_start and warehouse_course_end are very similar. Could they be refactored as a util function?

Reply:
Yep, I gave it a go [here](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/666/commits/a0fa6b6d9f37bdbec3c7cf604f53a9949ab43af4).